### PR TITLE
[examples, book] adds ID Pointer pattern example

### PR DESCRIPTION
**Pull request by damirka**

### CODE REVIEW

**Review by amnn — APPROVED** [review]
Looks good to my relatively untrained eyes!

File: doc/book/src/patterns/id-pointer.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# ID Pointer
+
+ID Pointer is a technique which allows separation of the main data (an object) and its accessors / capabilities by linking the latter to the original. There's a few different directions in which this pattern can be used:
```
Comment (randall-Mysten) [inline doc/book/src/patterns/id-pointer.md:3]:
```suggestion
ID Pointer is a technique that separates the main data (an object) and its accessors / capabilities by linking the latter to the original. There's a few different directions in which this pattern can be used:
```

Is there a different word for accessors? And it is odd to say that it separates the data by linking the things together. Is isolate correct? Is it that ID Pointer creates a link between objects and accessors?

File: doc/book/src/patterns/id-pointer.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- issuing transferable capabilities for shared objects (for example, a TransferCap that changes 'owner' field of a shared object)
+- splitting dynamic data and static (eg NFT and its Collection information)
```
Comment (randall-Mysten) [inline doc/book/src/patterns/id-pointer.md:6]: dynamic data and static data?
```suggestion
- splitting dynamic data and static (for example, an NFT and its Collection information)
```
why is Collection upper case? The way this is worded the NFT is the dynamic data?

File: doc/book/src/patterns/id-pointer.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- avoiding unnecessary type linking (and witness requirement) in generic applications (LP token for a LiquidityPool)
+
+```move
+{{#include ../../examples/sources/patterns/id_pointer.move:4:}}
+```
+
+This pattern is used in these examples:
```
Comment (randall-Mysten) [inline doc/book/src/patterns/id-pointer.md:13]:
```suggestion
The following examples demonstrate using this pattern:
```

**Review by randall-Mysten — COMMENTED** [review]
Added a few suggestions

File: doc/book/src/patterns/id-pointer.md (hunk 2 above)
Comment (damirka) [reply doc/book/src/patterns/id-pointer.md]: Uppercase because it's a Type.

File: doc/book/src/patterns/id-pointer.md (hunk 3 above)
Comment (damirka) [reply doc/book/src/patterns/id-pointer.md:13]: I'll keep it as is to align with everything else (on every other page we have this wording).

**Pull request merged**